ch02/tacos/src/main/java/tacos/Taco.java

```diff
@@ -12,15 +12,17 @@
 public class Taco {
 
   // end::allButValidation[]
-  @NotNull
+  //@NotNull
```

Comment:
Any reason you have left in the original versions as comments? I would remove these.